python/main.py

```diff
@@ -0,0 +1,166 @@
+
+
+# Load .env
+load_dotenv()
+
+# Constants setup
+POSTGRES_CREDENTIALS = {
+    "host": os.environ.get("POSTGRES_HOST"),
+    "port": int(os.environ.get("POSTGRES_PORT")),
+    "database": os.environ.get("POSTGRES_DB"),
+    "user": os.environ.get("POSTGRES_LOGIN"),
+    "password": os.environ.get("POSTGRES_PASSWORD"),
+}
+
+MINIO_CREDENTIALS = {
+    "endpoint": os.environ.get("MINIO_ENDPOINT"),
+    "access_key": os.environ.get("MINIO_ACCESS_KEY"),
+    "secret_key": os.environ.get("MINIO_SECRET_KEY"),
+}
+
+BUCKET_NAME = os.environ.get("BUCKET_NAME", "documents")
+
+# DB connect
+try:
+    print("Connecting to DB '{}' with user '{}'...".format(
+        POSTGRES_CREDENTIALS["database"], POSTGRES_CREDENTIALS["user"]))
+    conn = psycopg2.connect(**POSTGRES_CREDENTIALS)
+    print("Connection SUCCESS\n")
+except Exception as error:
+    print(error)
+    exit()
+
+cur = conn.cursor()
+
+# MinIO Connect
+try:
+    print("Connecting to MinIO server...")
+    client = Minio(
+        **MINIO_CREDENTIALS,
+        secure=False
+    )
+    print("Connection SUCCESS\n")
+except S3Error as error:
+    print(error)
+    exit()
+
+# Bucket create/retrieve
+found = client.bucket_exists(BUCKET_NAME)
+if not found:
+    print("Bucket '{}' does not exist yet, creating one...".format(BUCKET_NAME))
+    client.make_bucket(BUCKET_NAME)
+    print("Bucket '{}' successfully created\n".format(BUCKET_NAME))
+else:
+    print("Bucket '{}' already exists".format(BUCKET_NAME))
+    print("Removing all items from bucket '{}'...".format(BUCKET_NAME))
+    for item in client.list_objects(bucket_name=BUCKET_NAME):
+        client.remove_object(BUCKET_NAME, item.object_name)
+    print("Bucket '{}' has been successfully flushed !\n".format(BUCKET_NAME))
+
+
+# Bucket policy management
+# Anonymous read-only bucket policy
+policy = {
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": "*",
+            "Action": ["s3:GetBucketLocation", "s3:ListBucket"],
+            "Resource": "arn:aws:s3:::{}".format(BUCKET_NAME),
+        },
+        {
+            "Effect": "Allow",
+            "Principal": "*",
+            "Action": ["s3:GetObject"],
+            "Resource": "arn:aws:s3:::{}/*".format(BUCKET_NAME),
+        },
+    ],
+}
+
+try:
+    print("Setting bucket '{}' policy to 'PUBLIC' & 'READ ONLY'".format(BUCKET_NAME))
+    client.set_bucket_policy(BUCKET_NAME, json.dumps(policy))
+    print("Bucket '{}' policy successfully updated\n".format(BUCKET_NAME))
+except S3Error as error:
+    print(error)
+
+
+# Setup timeline_documents table
+try:
+    print("Creating 'timeline_documents' table in '{}' database...".format(
+        POSTGRES_CREDENTIALS["database"]))
+    cur.execute(
+        "CREATE TABLE timeline_documents (patient_id INT PRIMARY KEY NOT NULL, doc_name VARCHAR(50), "
+        "doc_url VARCHAR(100), CONSTRAINT fk_patient FOREIGN KEY (patient_id) REFERENCES osiris_patient(patient_id))")
```

Comment:
this is not the pythonic way of doing things, we can see this together

Reply:
I'm up for it